File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buyingStock = Integer.MAX_VALUE;
        int sellingStock = Integer.MIN_VALUE;
        int profit = 0;
        for (int i = 0; i < prices.length; i++) {
            if (buyingStock > prices[i]) {
                buyingStock = prices[i];
                System.out.println("buyingStock otto" + buyingStock);
            } else {
                int currentProfit = prices[i] - buyingStock;
                if (currentProfit > profit) {
                    profit = currentProfit;
                }
            }
            System.out.println("buyingStock i" + profit);
        }
        return profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # mini = prices[0]
        # profit = 0
        # for i in range(1,len(prices)):
        #     cost = prices[i] - mini
        #     profit = max(profit,cost)
        #     mini = min(mini,prices[i])
        # return profit
        min_price = prices[0]
        # print(min_price)
        profit = 0
        for i in range(1,len(prices)):
            profit = max(profit,prices[i] - min_price)
            min_price = min(min_price,prices[i])
        return profit
            # profit = max(profit,cost)
        
        